java/arcs/sdk/android/storage/service/StorageService.kt

```diff
@@ -80,37 +88,66 @@ open class StorageService : ResurrectorService() {
         ProtoPrefetcher.prefetch()
         StoreWriteBack.init(writeBackScope)
 
+        schedulePeriodicJobs(config)
+    }
+
+    private fun scheduleTtlJob(ttlHoursInterval: Long) {
         val periodicCleanupTask =
-            PeriodicWorkRequest.Builder(PeriodicCleanupTask::class.java, 12, TimeUnit.HOURS)
-                .addTag(PeriodicCleanupTask.WORKER_TAG)
-                .build()
-        val garbageCollectionTask =
             PeriodicWorkRequest.Builder(
-                DatabaseGarbageCollectionPeriodicTask::class.java,
-                1,
-                TimeUnit.DAYS
+                PeriodicCleanupTask::class.java,
+                ttlHoursInterval,
+                TimeUnit.HOURS
             )
-                .addTag(DatabaseGarbageCollectionPeriodicTask.WORKER_TAG)
-                .setConstraints(
-                    Constraints.Builder()
-                        .setRequiresDeviceIdle(true)
-                        .setRequiresCharging(true)
-                        .build()
-                ).build()
-
-        val workManager = WorkManager.getInstance(this)
+            .addTag(PeriodicCleanupTask.WORKER_TAG)
+            .build()
         workManager.enqueueUniquePeriodicWork(
             PeriodicCleanupTask.WORKER_TAG,
-            ExistingPeriodicWorkPolicy.KEEP,
+            ExistingPeriodicWorkPolicy.REPLACE,
             periodicCleanupTask
         )
+    }
+
+    private fun scheduleGcJob(garbageCollectionHoursInterval: Long) {
+        val garbageCollectionTask =
+            PeriodicWorkRequest.Builder(
+                DatabaseGarbageCollectionPeriodicTask::class.java,
+                garbageCollectionHoursInterval,
+                TimeUnit.HOURS
+            )
+            .addTag(DatabaseGarbageCollectionPeriodicTask.WORKER_TAG)
+            .setConstraints(
+                Constraints.Builder()
+                    .setRequiresDeviceIdle(true)
+                    .setRequiresCharging(true)
+                    .build()
+            ).build()
         workManager.enqueueUniquePeriodicWork(
             DatabaseGarbageCollectionPeriodicTask.WORKER_TAG,
-            ExistingPeriodicWorkPolicy.KEEP,
+            ExistingPeriodicWorkPolicy.REPLACE,
             garbageCollectionTask
         )
     }
 
+    protected fun schedulePeriodicJobs(config: StorageServiceConfig) {
+        if (!config.ttlJobEnabled) {
+            workManager.cancelAllWorkByTag(PeriodicCleanupTask.WORKER_TAG)
+        } else {
+            scheduleTtlJob(config.ttlHoursInterval)
+        }
```

Comment:
Nit: flip the if-condition and branches around so that you're checking a positive condition rather than a negative one.

Same for the if-statement below.